Bio/SeqFeature.py

```diff
@@ -851,6 +851,16 @@ def __iter__(self):
             for i in range(self._start, self._end):
                 yield i
 
+    def __eq__(self, other):
+        """Check if start, end and strand of this location are the same as in the other location"""
+        if not isinstance(other, FeatureLocation):
+            return False
+        return self._start == other.start and self._end == other.end and self._strand == other.strand
```

Comment:
I'd just check the attributes of the parts of the ``CompoundLocation`` for that CDS (or a similar feature) as created by the GenBank parser. For example the ``2201..2479`` bit should have ``.ref`` and ``.ref_db`` equal to ``None``, while the ``U18267.1:120..246`` bit would I think have ``.ref`` equal to ``U18267.1`` and ``.ref_db`` still ``None``.

(This bit of ``Bio.SeqFeature`` seems to have been designed with a bit more scope than is currently used in the GenBank/EMBL parsers)